test/e2e/operator.go

```diff
@@ -436,18 +436,26 @@ var _ = Describe("ARO Operator - ImageConfig Reconciler", func() {
 
 	// Reimplementation of function from image config controller
 	getCloudAwareRegistries := func(instance *arov1alpha1.Cluster) ([]string, error) {
```

Comment:
Do we need this function reimplemented or is it possible to import the one from `pkg/operator`?

Reply:
It's a good thought but there are a couple problems with it:

1. The function is in the `imageconfig` controller which would need to be imported here, I'd like to keep the e2e test as separate from the controller as possible to ensure it's truly e2e.

1. The `getCloudAwareRegistries` function isn't exported by the controller and changing that for the purposes of the test didn't seem like a good solution either.

To me the problem is there's no one place to go to get the required ACR endpoints for a cluster. There used to just be "arosvc.azurecr.io" but that's grown to include USGovCloud ("arosvc.azurecr.us"), replication registries ("arosvc.<region>.data.azurecr.io"), and separate INT environments for each ("arosvc" → "arointsvc"). A better solution would be creating a way for the RP to tell the cluster which registries are required on creation, but we don't currently have that and it would require more time and effort to do than just copying and pasting this code for now. But it is something we _should_ get on our backlog now before we get even more possibilities to account for and these functions balloon even more.

Reply:
> The `getCloudAwareRegistries` function isn't exported by the controller and changing that for the purposes of the test didn't seem like a good solution either

Any particular reason why it can't be exported?

Reply:
Not a technical one. I was hesitant to do export it just for the purposes of the test but given the feedback it seems others don't share my concerns, which is fine.

I still think a better long-term solution would be for a cluster to be configured with all its required registries by the RP at install time. We've been doing that with `arosvc.azurecr.io` when it was the only ACR we were concerned with, but the platform has grown.